3p/ampcontext.js

```diff
@@ -62,24 +110,38 @@ export class AmpContext {
    *    every time we receive a page visibility message.
    */
   observePageVisibility(callback) {
-    return this.client_.makeRequest(
-        MessageType.SEND_EMBED_STATE,
-        MessageType.EMBED_STATE,
-        callback);
+    const visibilityCallbacksLength = this.visibilityCallbacks_.length;
```

Comment:
- Removed handler from `3p/integration.js`
- Added handler (`dispatchVisibilityChangeEvent_`) in `AmpContext.js`